Backend/app/rider-platform/rider-app/Main/src/Environment.hs

```diff
@@ -294,11 +303,26 @@ instance Registry Flow where
   registryLookup = Registry.withSubscriberCache performLookup
     where
       performLookup sub = do
-        fetchFromDB sub.merchant_id >>= \registryUrl -> do
+        fetchFromDB (Id sub.merchant_id) >>= \registryUrl -> do
           Registry.registryLookup registryUrl sub
+            `catch` \e -> handleRegistryError e registryUrl sub
+      fetchFromDB :: Id DM.Merchant -> Flow BaseUrl
       fetchFromDB merchantId = do
-        merchant <- CM.findById (Id merchantId) >>= fromMaybeM (MerchantDoesNotExist merchantId)
-        pure $ merchant.registryUrl
+        merchant <- CM.findById merchantId >>= fromMaybeM (MerchantDoesNotExist merchantId.getId)
+        registryUrl <-
+          case merchant.gatewayAndRegistryPriorityList of
+            (NY : _) -> asks (.nyRegistryUrl)
+            _ -> asks (.ondcRegistryUrl)
+        pure registryUrl
+      handleRegistryError :: ExternalAPICallError -> BaseUrl -> SimpleLookupRequest -> Flow (Maybe Subscriber)
+      handleRegistryError _ registryUrl sub = do
+        merchant <- CM.findById (Id sub.merchant_id) >>= fromMaybeM (MerchantDoesNotExist sub.merchant_id)
+        let networkPriorityList = reorderList merchant.gatewayAndRegistryPriorityList
+        CM.updateGatewayAndRegistryPriorityList merchant networkPriorityList
```

Comment:
Let's not have this logic here. We can make use of failover API to do the switching which can be called based on prom metrics.

Reply:
Here should we call performLookup withRetryConfig and check with the other registry?
And then call the update function based on metrics